src/SQONBuilder.ts

```diff
@@ -261,18 +261,22 @@ const createBuilder = (sqon: SQON): SQONBuilder => {
 	};
 };
 
-const combine = (op: CombinationKey, sqon: SQON, content: SQON | SQON[]): CombinationOperator => {
-	if (sqon.op === op) {
-		return {
-			op,
-			content: [...sqon.content, ...asArray(content)],
-		};
+const combine = (op: CombinationKey, sqon: SQON, content: SQON | SQON[], pivot?: string): CombinationOperator => {
+	const output: CombinationOperator = { op, content: [] };
+	if (sqon.op === op && sqon.pivot === pivot) {
+		// provided sqon has same operation as requested:
+		//   don't add wrapper, insert new content into existing sqon
+		output.content = output.content.concat(sqon.content, content);
 	} else {
-		return {
-			op,
-			content: [sqon, ...asArray(content)],
-		};
+		output.content = output.content.concat(sqon, content);
+	}
+
+	if (pivot !== undefined) {
+		// dont automatically assign pivot: undefined to an operator, it will then appear in every output and that is not desired
+		// optionally assigning pivot only if it has a value avoids this
+		output.pivot = pivot;
```

Comment:
You've seen this before... it's because the chai assertion for `deep.equal` believes `{ a:1, b: undefined }` and `{ a:1 }` are not equal. In usage, however, all outputs with `pivot: undefined` in the SQON output the exact same... ie. `toString` and `toValue` do not output with `pivot: undefined`. This is annoyingly only here to placate the testing library.

The only functional difference I can see right now is that it would indicate the SQON `hasOwnProperty` of `pivot` which... is meaningless since the value of the sqon.pivot will be undefined whether there or not.

Reply:
hm you're right. I'm not excited about this but I can see how tool is throwing itself in the engine. 

please add a note to remind us to revisit this later?